lib/DB/GeophotoMapper.php

```diff
@@ -12,61 +12,200 @@
+	 * @throws \OCP\DB\Exception
+	 */
+    public function findByFileIdUserId($fileId, $userId) {
+		$qb = $this->db->getQueryBuilder();
+
+		$qb->select("*")
+			->from($this->getTableName())
+			->where(
+				$qb->expr()->eq('user_id', $qb->createNamedParameter($userId, IQueryBuilder::PARAM_STR))
+			)->andWhere(
+				$qb->expr()->eq('file_id', $qb->createNamedParameter($fileId, IQueryBuilder::PARAM_STR))
+			);
+
+		return $this->findEntity($qb);
     }
 
+	/**
+	 * @param $fileId
+	 * @return mixed|\OCP\AppFramework\Db\Entity
+	 * @throws \OCP\AppFramework\Db\DoesNotExistException
+	 * @throws \OCP\AppFramework\Db\MultipleObjectsReturnedException
+	 * @throws \OCP\DB\Exception
+	 */
+	public function findByFileId($fileId) {
+		$qb = $this->db->getQueryBuilder();
+
+		$qb->select("*")
+			->from($this->getTableName())
+			->where(
+				$qb->expr()->eq('file_id', $qb->createNamedParameter($fileId, IQueryBuilder::PARAM_STR))
+			);
+
+		return $this->findEntity($qb);
+	}
+
+	/**
+	 * @param $userId
+	 * @param $limit
+	 * @param $offset
+	 * @return array|\OCP\AppFramework\Db\Entity[]
+	 * @throws \OCP\DB\Exception
+	 */
     public function findAll($userId, $limit=null, $offset=null) {
-        $sql = 'SELECT * FROM `*PREFIX*maps_photos` where `user_id` = ? and `lat` is not null and `lng` is not null ORDER BY `date_taken` ASC';
-        return $this->findEntities($sql, [$userId], $limit, $offset);
+		$qb = $this->db->getQueryBuilder();
+
+		$qb->select("*")
+			->from($this->getTableName())
+			->where(
+				$qb->expr()->eq('user_id', $qb->createNamedParameter($userId, IQueryBuilder::PARAM_STR))
+			)->andWhere(
+				$qb->expr()->isNotNull('lat')
+			)->andWhere(
+				$qb->expr()->isNotNull('lng')
+			)->orderBy('date_taken', 'ASC');
+		if (!is_null($offset)) {
+			$qb->setFirstResult($offset);
+		}
+		if (!is_null($limit)) {
+			$qb->setMaxResults($limit);
+		}
+		return $this->findEntities($qb);
     }
 
+	/**
+	 * @param $userId
+	 * @param $limit
+	 * @param $offset
+	 * @return array|\OCP\AppFramework\Db\Entity[]
+	 * @throws \OCP\DB\Exception
+	 */
     public function findAllNonLocalized($userId, $limit=null, $offset=null) {
-        $sql = 'SELECT * FROM `*PREFIX*maps_photos` where `user_id` = ? and (`lat` is null or `lng` is  null) ORDER BY `date_taken` ASC';
-        return $this->findEntities($sql, [$userId], $limit, $offset);
+		$qb = $this->db->getQueryBuilder();
+
+		$qb->select("*")
+			->from($this->getTableName())
+			->where(
+				$qb->expr()->eq('user_id', $qb->createNamedParameter($userId, IQueryBuilder::PARAM_STR))
+			)->andWhere(
+				$qb->expr()->orX(
+					$qb->expr()->isNull('lat'),
+					$qb->expr()->isNull('lng')
+				)
+			)->orderBy('date_taken', 'ASC');
+		if (!is_null($offset)) {
+			$qb->setFirstResult($offset);
+		}
+		if (!is_null($limit)) {
+			$qb->setMaxResults($limit);
+		}
+		return $this->findEntities($qb);
     }
 
+	/**
+	 * @param $fileId
+	 * @return false|void
```

Comment:
executeStatement always returns int

```suggestion
	 * @return int
```